src/main/java/networkbook/logic/LogicManager.java

```diff
@@ -50,6 +51,10 @@ public CommandResult execute(String commandText) throws CommandException, ParseE
         Command command = networkBookParser.parseCommand(commandText);
         commandResult = command.execute(model);
 
+        if (command instanceof DoesNotChangeDataCommand) {
+            return commandResult;
+        }
```

Comment:
Would it be better if changing command is a method that returns a value of a field, i.e.
```java
private boolean isChangingData;

public boolean isChangingData() {
    return this.isChangingData;
}
```
rather than having two separate classes for it? I do not see `ChangeDataCommand` class being in good use, and the current implementation does not force concrete classes to inherit either from `ChangeDataCommand` class or `DoesNotChangeDataCommand` class.

Reply:
Okay makes sense, will change to your suggested implementation, with a constructor in `Command` class that must be called from the child classes.